Add new iptables plugin

ririsoft opened the pull request:
### Required for all PRs:

- [ ] CHANGELOG.md updated
- [X] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [X] README.md updated (if adding a new plugin)

The iptables plugin aims at monitoring bytes and packet counters
matching a given set of iptables rules.

Typically the user would set a dedicated monitoring chain into a given
iptables table, and add the rules to monitor to this chain. The plugin
will allow to focus on the counters for this particular table/chain.

ririsoft commented:
Hello,
Where should I put the changelog info, everything looks ready for 1.0 release ... Should I put it there too ?
Cheers.

phemmer commented on plugins/inputs/iptables/iptables.go:
```diff
@@ -0,0 +1,131 @@
+		if len(mv) == 0 || len(mv[0]) != 5 {
+			continue
+		}
+		tags := map[string]string{"table": ipt.Table, "chain": mchain[1], "ruleid": strconv.Itoa(i + 1)}
```
I don't think using the line number as an identifier tag is a good idea. From the perspective of someone looking at the data in influxdb (or wherever the telegraf data is going), `ruleid=4` is going to mean very next to nothing. Also, if another iptables rule is inserted, then line 4 becomes line 5, and tracking the metrics gets screwed up. It may also not going to be consistent between hosts (one host's rule 4 may not be the same as another host's rule 4).

ririsoft replied:
I agree. I am using comments exclusively myself to identify the rules. I considered using the rule number as a fallback for rules that do not have a comment. For the sake of clarify I suggest to ignore rules that do not have a comment. What do you think ? Any other suggestion to distinguish a rule from another ?

sparrc replied:
@ririsoft what do these lines look like? is it common not to have a comment?

ririsoft replied:
@sparrc: I have no idea if it is common or not. I can add an option to let the user ignore rules without comment or not as he prefer. If rules without comment are to be included then the ruleid will be the rule number. What do you think ?

sparrc replied:
I'm not sure the "rule id" should ever be used, that seems like it can only lead to confusion among users, and can also lead to bad data when rules are added/deleted. If there is no way to uniquely identify it then I think it's best to skip it.

phemmer commented on plugins/inputs/iptables/README.md:
```diff
@@ -0,0 +1,48 @@
+The iptables plugin gathers packets and bytes counters for rules within a set of table and chain from the Linux's iptables firewall.
+
+The iptables command requires root access. You can run telegraf as root (not recommended), or you
+can configure telegraf to use `sudo` for running iptables.
```
...or you can use [CAP_NET_ADMIN](http://linux.die.net/man/7/capabilities)

ririsoft replied:
Excellent, I will mention this in the doc/config. While sudo allows to grant less rights, CAP_NET_ADMIN makes it easier to configure through the systemd service configuration file. Thanks !

ririsoft commented:
I am preparing a new version of the plugin that will:
* Discard rules without a comment
* Update the documentation with information for using systemd capabilities to run the plugin without the need for root or sudo. It will let the possibility to run iptables through sudo though, for users who do not want to give too much capabilities to telegraf.

It will be available next week.

sparrc commented:
sounds great, thank you @ririsoft

ririsoft commented:
Pull request is now completed. The failing CI does not look related to this development.
Do not hesitate to ask for a rebase and/or squash and if you would like an update in the changelog too.
Cheers.

sparrc commented:
OK, thanks @ririsoft, can you:

1. rebase your changes
2. update the CHANGELOG (under release 1.1)
3. add your plugin to the list of available plugins on the main repo README?

ririsoft commented:
Sorry I messed up with the CHANGELOG.md but it should be ok now: I rebased and squashed all the porcelain. Thank you very much for your kind help on this PR.
